webapp/channels/src/components/integrations/installed_incoming_webhooks/index.ts

```diff
@@ -4,34 +4,35 @@
 import {connect} from 'react-redux';
 import {bindActionCreators} from 'redux';
 import type {Dispatch} from 'redux';
-
 import type {GlobalState} from '@mattermost/types/store';
 
 import {removeIncomingHook} from 'mattermost-redux/actions/integrations';
 import {Permissions} from 'mattermost-redux/constants';
 import {getAllChannels} from 'mattermost-redux/selectors/entities/channels';
 import {getConfig} from 'mattermost-redux/selectors/entities/general';
-import {getIncomingHooks} from 'mattermost-redux/selectors/entities/integrations';
 import {haveITeamPermission} from 'mattermost-redux/selectors/entities/roles';
 import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
 import {getUsers} from 'mattermost-redux/selectors/entities/users';
 
 import {loadIncomingHooksAndProfilesForTeam} from 'actions/integration_actions';
 
 import InstalledIncomingWebhooks from './installed_incoming_webhooks';
+import {getIncomingHooks, getIncomingHooksTotalCount} from 'mattermost-redux/selectors/entities/integrations';
 
 function mapStateToProps(state: GlobalState) {
-    const config = getConfig(state);
     const teamId = getCurrentTeamId(state);
+    const incomingHooksFromState = getIncomingHooks(state);
+    const incomingHooks = Object.keys(incomingHooksFromState)
```

Comment:
`incomingHooks` needs to be created in a memoized selector using `createSelector` or else it'll return a new array every time it runs which will then cause the component to re-render constantly

Reply:
Good catch. I implemented it as suggested.